src/main/java/uk/gov/justice/digital/delius/service/OffenderService.java

```diff
@@ -137,17 +142,21 @@ public Long getOffenderCount() {
 
     }
 
-    // TODO DT-337 Flesh out this stub
     @Transactional(readOnly = true)
     public OffenderLatestRecall getOffenderLatestRecall(Long offenderId) {
-        offenderRepository.findByOffenderId(offenderId)
-                .map(offender -> convictionService.getActiveCustodialEvent(offender.getOffenderId()))
-                .map(activeCustodialEvent -> {
-                    return Optional.ofNullable(activeCustodialEvent.getDisposal())
-                            .map(Disposal::getCustody)
-                            .orElseThrow(() -> new CustodyNotFoundException(activeCustodialEvent));
-                })
-                .orElseThrow(() -> new NotFoundException("Offender not found"));
-        return null;
+        Event actualCustodialEvent = convictionService.getActiveCustodialEvent(offenderId);
+        Custody custody = findCustodyOrThrow(actualCustodialEvent);
+        return custody.findLatestRelease()
+                .map(releaseTransformer::offenderLatestRecallOf)
+                .orElse(OffenderLatestRecall.NO_RELEASE);
+    }
+
+    private Custody findCustodyOrThrow(Event activeCustodialEvent) {
+        return Optional.ofNullable(activeCustodialEvent.getDisposal())
+                .filter(not(Disposal::isSoftDeleted))
```

Comment:
I think we should also add this filtering to `convictionService.getActiveCustodialEvent` if not there already.

Reply:
This already occurs in ConvictionService#activeCustodyEvents.